runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1224,17 +1652,11 @@ TR::Register *J9::Power::TreeEvaluator::awrtbariEvaluator(TR::Node *node, TR::Co
    return NULL;
    }
 
-TR::Register *J9::Power::TreeEvaluator::irdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-   {
-   return TR::TreeEvaluator::irdbariEvaluator(node, cg);
-   }
-
-TR::Register *J9::Power::TreeEvaluator::irdbariEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+TR::Register *iGenerateSoftwareReadBarrier(TR::Node *node, TR::CodeGenerator *cg)
```

Comment:
Maybe a better name would be `compressedObjLoadWithReadBarrier`. On x they use `performHeapLoadWithReadBarrier` but we can't use that because we have two different versions for compressed and uncompressed objects.

Reply:
See related review [1]. If anything we need to consolidate the name on all platforms. The two different evaluators for `a` and for `i` seem > 80% equivalent judging from a quick diff between them. We can likely consolidate both into a single `TR::TreeEvaluator::generateSoftwareReadBarrier`.

Thinking about this, why do we have the `i` evaluator for concurrent scavenge at all? Shouldn't it always be `a` given we want to evacuate collected references?

[1] https://github.com/eclipse/openj9/pull/6234#discussion_r296506228

Reply:
I don't expect any such CS related changes to be made here, as it is outside the scope of this PR. Perhaps we can continue discussion in a separate issue?

Reply:
The high level is under CS `irdbari` is generated in the trees when loading the object under compressedref. `ardbari` is generated in the trees when loading the object under noncompressedrefs. Going deeper then this is probably better left for a separate issue.